.pipelines/templates/.builder-release-template-windows.yaml

```diff
@@ -5,10 +5,11 @@ parameters:
 
 steps:
   - bash: |
-      if [[ -n ${SIG_GALLERY_NAME} && -n ${SIG_IMAGE_NAME_PREFIX} && -n ${SIG_IMAGE_VERSION} ]]; then m="sigMode"; sigImageName="${SIG_IMAGE_NAME_PREFIX}-${WINDOWS_SKU}"; else m="default"; fi && \
+      if [[ $(HYPERV_GENERATION) == "V2" ]]; then m="gen2Mode"; elif [[ -n ${SIG_GALLERY_NAME} && -n ${SIG_IMAGE_NAME_PREFIX} && -n ${SIG_IMAGE_VERSION} ]]; then m="sigMode"; sigImageName="${SIG_IMAGE_NAME_PREFIX}-${WINDOWS_SKU}"; else m="default"; fi && \
```

Comment:
I think that we also need to support `sigMode` in building WS2022 gen2 VHD for test purpose. I think that there are below options:
1. set mode to `sigMode` with `HYPERV_GENERATION=V2`
2. set mode to `gen2Mode` with setting `SIG_GALLERY_NAME`, `sigImageName` and `SIG_IMAGE_VERSION`
3. define a new mode `gen2SigMode`

Reply:
For `HYPERV_GENERATION=V2`, we can define a new mode `gen2SigMode` if `SIG_GALLERY_NAME`, `sigImageName` and `SIG_IMAGE_VERSION` are set. If they are not set, mode is just `gen2Mode`? This will only add one condition branch in `packer.mk`: `ifeq (${MODE},gen2SigMode)`?

Reply:
let us sync in teams

Reply:
Decision: we will reuse `sigMode` for Gen 2 and use the variable "`GEN2_SIG_FOR_PRODUCTION`" to indicate if it's for test or for publishing. If pipeline variables `SIG_GALLERY_NAME`, `SIG_IMAGE_NAME_PREFIX`, and `SIG_IMAGE_VERSION` are set, we are using sig for test. O.w. we use sig for production.